### PR TITLE
codeisn extension safe iOS framework

### PR DESCRIPTION
**Pull request by cyanglaz**
Update the conductor to code sign extension safe iOS frameworks
The new frameworks are added in https://github.com/flutter/engine/pull/45781/files

part of https://github.com/flutter/flutter/issues/124291

Also includes an Engine roll includes:
https://github.com/flutter/engine/commit/9232e76c30455ee1c0c1584001bdbb01000f8212
https://github.com/flutter/engine/commit/39c0f2ea1f536eb7dc057d9147f23e3fec4a56c9

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by christopherfujino** [comment]
I believe this change needs to ALSO roll the engine, or else tests will fail the other way

**Comment by cyanglaz** [comment]
> I believe this change needs to ALSO roll the engine, or else tests will fail the other way

I see, what was the process before?

**Review by christopherfujino — APPROVED** [review]
LGTM, pending the conductor unit tests pass of course

**Comment by christopherfujino** [comment]
`Mac framework_test_misc` failed again, but it looks unrelated?

```
00:10 +191 ~2: /Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart: group retry flag allows test to run multiple times the group with retry flag the test inside it    
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: true
  Actual: <false>

When the exception was thrown, this was the stack:
#4      main.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart:54:9)
#5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:168:29)
<asynchronous suspension>
#6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1013:5)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart line 54
The test description was:
  the test inside it
```

**Comment by christopherfujino** [comment]
> `Mac framework_test_misc` failed again, but it looks unrelated?
> 
> ```
> 00:10 +191 ~2: /Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart: group retry flag allows test to run multiple times the group with retry flag the test inside it    
> ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
> The following TestFailure was thrown running a test:
> Expected: true
>   Actual: <false>
> 
> When the exception was thrown, this was the stack:
> #4      main.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart:54:9)
> #5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:168:29)
> <asynchronous suspension>
> #6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1013:5)
> <asynchronous suspension>
> <asynchronous suspension>
> (elided one frame from package:stack_trace)
> 
> This was caught by the test expectation on the following line:
>   file:///Volumes/Work/s/w/ir/x/w/flutter/packages/flutter_test/test/widget_tester_test.dart line 54
> The test description was:
>   the test inside it
> ```

Oh wait, never mind, that doesn't look like it actually added to the test failures...

**Comment by christopherfujino** [comment]
This is the real failure:

```
01:16 +18: test/codesign_integration_test.dart: validate the expected binaries from the conductor codesign command are present in the cache                                                            

Codesign integration test failed.

This means that the binary files found in the Flutter cache do not match those
expected by the conductor tool (either an expected file was not found in the
cache or an unexpected file was found in the cache).

This usually happens either during an engine roll or a change to the caching
logic in flutter_tools. If this is a valid change, then the conductor source
code should be updated, specifically either the [binariesWithEntitlements] or
[binariesWithoutEntitlements] lists, depending on if the file should have macOS
entitlements applied during codesigning.


01:16 +18 -1: test/codesign_integration_test.dart: validate the expected binaries from the conductor codesign command are present in the cache [E]                                                     
  Found unexpected binary in cache: /Volumes/Work/s/w/ir/x/t/flutter_conductor_integration_test.24RS1b/flutter_conductor_checkouts/framework/bin/cache/artifacts/engine/ios-profile/extension_safe/Flutter.framework/Flutter
  package:matcher                           fail
  test/codesign_integration_test.dart 72:7  main.<fn>

```

File: dev/conductor/core/lib/src/codesign.dart [hunk 1]
```diff
@@ -201,10 +201,16 @@ class CodesignCommand extends Command<void> {
       'artifacts/engine/darwin-x64/libtessellator.dylib',
       'artifacts/engine/ios-profile/Flutter.xcframework/ios-arm64/Flutter.framework/Flutter',
       'artifacts/engine/ios-profile/Flutter.xcframework/ios-arm64_x86_64-simulator/Flutter.framework/Flutter',
+      'artifacts/engine/ios-profile/extension_safe/Flutter.xcframework/ios-arm64/Flutter.framework/Flutter',
```
Comment (christopherfujino) [inline dev/conductor/core/lib/src/codesign.dart:204]: this should be xcframework, not .framework:

```
01:16 +18 -1: test/codesign_integration_test.dart: validate the expected binaries from the conductor codesign command are present in the cache [E]                                                     
  Found unexpected binary in cache: /Volumes/Work/s/w/ir/x/t/flutter_conductor_integration_test.24RS1b/flutter_conductor_checkouts/framework/bin/cache/artifacts/engine/ios-profile/extension_safe/Flutter.framework/Flutter
  package:matcher                           fail
  test/codesign_integration_test.dart 72:7  main.<fn>

```

**Comment by cyanglaz** [comment]
I temporarily updated the engine version to https://github.com/flutter/engine/pull/46014/commits/76463ab1c9eaa265e2355c6fab2e7b4bd71ba6a9. This should be changed to a post-submit commit when https://github.com/flutter/engine/pull/46014 lands

**Comment by christopherfujino** [comment]
> I temporarily updated the engine version to [flutter/engine@76463ab](https://github.com/flutter/engine/commit/76463ab1c9eaa265e2355c6fab2e7b4bd71ba6a9). This should be changed to a post-submit commit when [flutter/engine#46014](https://github.com/flutter/engine/pull/46014) lands

that won't test anything, as there won't be any binaries on cloud storage.

**Comment by christopherfujino** [comment]
> > I temporarily updated the engine version to [flutter/engine@76463ab](https://github.com/flutter/engine/commit/76463ab1c9eaa265e2355c6fab2e7b4bd71ba6a9). This should be changed to a post-submit commit when [flutter/engine#46014](https://github.com/flutter/engine/pull/46014) lands
> 
> that won't test anything, as there won't be any binaries on cloud storage.

I updated your PR however.

**Comment by cyanglaz** [comment]
> that won't test anything, as there won't be any binaries on cloud storage.

I think there is a separate cloud storage bucket for pre-submit PR? @godofredoc

**Comment by godofredoc** [comment]
> > that won't test anything, as there won't be any binaries on cloud storage.
> 
> I think there is a separate cloud storage bucket for pre-submit PR? @godofredoc

To validate the change with the engine version you need to also update the realm file to instruct the tests to use the presubmit artifacts.

**Comment by christopherfujino** [comment]
> > > that won't test anything, as there won't be any binaries on cloud storage.
> > 
> > 
> > I think there is a separate cloud storage bucket for pre-submit PR? @godofredoc
> 
> To validate the change with the engine version you need to also update the realm file to instruct the tests to use the presubmit artifacts.

cool!

**Comment by christopherfujino** [comment]
I'm guessing it's this file? https://github.com/flutter/flutter/blob/master/bin/internal/engine.realm

**Comment by godofredoc** [comment]
> I'm guessing it's this file? https://github.com/flutter/flutter/blob/master/bin/internal/engine.realm

That's correct!

**Comment by cyanglaz** [comment]
@christopherfujino I think the codesign integration test passed? Should I land the engine PR fix (after tree turns green), then follow up with this one + a manual engine roll?

**Comment by christopherfujino** [comment]
> @christopherfujino I think the codesign integration test passed? Should I land the engine PR fix (after tree turns green), then follow up with this one + a manual engine roll?

SGTM, I would just let the engine sheriff know that if the autoroller rolls it, it will fail.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
Some of tests are wedged, I'm going to manually land it.

**Pull request merged**